themes/bootstrap3/templates/Recommend/ConsortialVuFind.phtml

```diff
@@ -0,0 +1,31 @@
+<div class="consortialVuFind">
+  <?php $results = $this->recommend->getResults() ?>
+  <?php $records = $results['records'] ?? [] ?>
+  <?php $resultCount = $results['resultCount'] ?? 0; ?>
+  <?php $moreResultsUrl = $this->recommend->getMoreResultsUrl(); ?>
+  <?php if (!empty($records)): ?>
+    <h2><?=$this->transEsc('consortial_vufind_recommend_heading')?></h2>
+    <?php if ($this->translate('consortial_vufind_recommend_intro', [], '')): ?>
```

Comment:
Is there a reason to call translate twice here? Couldn't you make the full translate call and assign that to a variable, and then check the variable to decide whether to display anything? Might be slightly more efficient.

Reply:
The issue here is how the user is supposed to disable the intro in the translation file.  Whether I comment out the translation line entirely or declare it with nothing after the equals sign, the result of calling `$this->translate` is a string that appears empty but is strlen = 3 (bytes).  So then there's no good check to decide whether to display it, except strlen(x) > 3 which is ugly, or some post-filtering for unprintable characters which would be unexpected.  Maybe I'm using translate wrong, or it has a bug.  Or there is some more complicated syntax for users to disable a translation key...

Reply:
Hmm, I can't remember, but it's possible that the translation system is defaulting to some multi-byte non-printing character as the default empty result for reasons that I can't currently remember. I'm curious if this variation would do the job correctly:

```
    <?php
      $intro = $this->translate('consortial_vufind_recommend_intro_html', [
          '%%result_count%%' => $this->localizedNumber($resultCount),
          '%%url%%' => $moreResultsUrl,
      ]);
    ?>
    <?php if ($intro): ?>
      <p><?=$intro?></p>
    <?php endif; ?>
```

If that's no good, let me know and I'll spend a little time refreshing my memory on what's going on and why.

Reply:
No luck with that variation.  $intro still has strlen = 3.  It appears to be [unicode 200c](https://unicode-explorer.com/c/200C), 'zero width non-joiner'.  trim() doesn't help.

Reply:
I'm looking into the en.ini loading.  in ExtendedIniReader->getTextDomain...

Reply:
Ah, it was intentional!  https://github.com/vufind-org/vufind/blob/53e1894abcbc4e8ef3ac1e629e4c161e197b672a/module/VuFind/src/VuFind/I18n/Translator/Loader/ExtendedIniReader.php#L85

Reply:
Yes, that's what I thought -- thanks for digging up the exact detail! But I'm still not sure why setting a default value of `''` causes things to work. If the translation system is translating the input key to the zero-width non-joiner, I wouldn't expect us to get a default value back. I'm also pretty sure we use this pattern elsewhere in the code -- but maybe it's not working right there either. Maybe it's worth just checking for empty or the zero-width non-joiner, since I think that's going to stick around.

Reply:
Done, with a comment to explain.